rosidl_typesupport_introspection_cpp/include/rosidl_typesupport_introspection_cpp/message_introspection.hpp

```diff
@@ -27,31 +27,68 @@
 namespace rosidl_typesupport_introspection_cpp
 {
 
+/// Structure used to describe a single field of an interface type.
 typedef struct ROSIDL_TYPESUPPORT_INTROSPECTION_CPP_PUBLIC MessageMember_s
 {
+  /// The name of the field.
   const char * name_;
+  /// The type of the field as a value of the field types enum,
+  /// e.g.rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT
   uint8_t type_id_;
+  /// If the field is a string, the upper bound on the length of the string.
   size_t string_upper_bound_;
+  /// If the type_id_ value is rosidl_typesupport_introspection_cpp::ROS_TYPE_MESSAGE
+  /// this points to an array describing the fields of the sub-interface.
   const rosidl_message_type_support_t * members_;
+  /// True if this field is an array, false if it is a unary type. An array has the same value for
+  /// type_id_.
   bool is_array_;
+  /// If is_array_ is true, this contains the number of members in the array.
   size_t array_size_;
+  /// If is_array_ is true, this specifies if the array has a maximum size. If it is true, the
+  /// value in array_size_ is the maximum size.
   bool is_upper_bound_;
+  /// The bytes into the interface's in-memory representation that this field can be found at.
   uint32_t offset_;
+  /// If the interface has a default value, this points to it.
   const void * default_value_;
+  /// If is_array_ is true, a pointer to a function that gives the size of array members.
+  /// First argument should be a pointer to the actual memory representation of the member.
+  /// Only used for nested interface types.
```

Comment:
Same comment about these functions being used for arrays of normal types too

```c++
  {
    "p",  // name
    ::rosidl_typesupport_introspection_cpp::ROS_TYPE_DOUBLE,  // type
    0,  // upper bound of string
    nullptr,  // members of sub message
    true,  // is array
    12,  // array size
    false,  // is upper bound
    offsetof(sensor_msgs::msg::CameraInfo, p),  // bytes offset in struct
    nullptr,  // default value
    size_function__CameraInfo__p,  // size() function pointer
    get_const_function__CameraInfo__p,  // get_const(index) function pointer
    get_function__CameraInfo__p,  // get(index) function pointer
    fetch_function__CameraInfo__p,  // fetch(index, &value) function pointer
    assign_function__CameraInfo__p,  // assign(index, value) function pointer
    nullptr  // resize(index) function pointer
  },
```

Reply:
Sorry, I did not understand the point. Is it that the comment is somehow wrong and you include the structure to demonstrate it? Should we include this structure in the comment? Another option?

Reply:
"Only used for nested interface types." sounds like it's saying these struct fields are only used if the type is `::ROS_TYPE_MESSAGE`, but they seem to be set for other array types too, like when `ROS_TYPE_DOUBLE` is set.

Reply:
aaah I think I get it. Removed the reference to nested interfaces 5ce14978aa84c236d6cd1aa7eb021c665112cb39